### ISSUE TITLE
New local files get deleted on 2-way sync

### ISSUE THREAD

**Issue by johnpbaldwin** [digexp/dxsync#6]
I am running into the following sequence:
1. I create a new file in my local machine
2. The file is recognized by dxsync and it says it uploaded it to the server.
3. The file does not exist on the server (when checked manually)
4. On the next 2-way synchronization, the local files are deleted.

Current resolution:
1. Stop dxsync
2. Create file on server
3. Start dxsync
4. File downloads
5. Proceed.

**Comment by jlanier** [comment]
Thanks John.. let me look into this and I'll get back to you.

**Comment by mburati** [comment]
John,   to help @jlanier  reproduce this more quickly,  can you let us know if this happens on all OSs (Linux, Mac, Windows) or only one of them (or at least which you're using),   which version of WebSphere Portal,   and which file(s) you've tried this with?

Thanks!

**Comment by johnpbaldwin** [comment]
I have seen this behavior in both Windows and Mac OS X.

I'm using WebSphere Portal 8.5 CF08.

When this has happened to me, I'm creating a new simple module. For example, I'll create a new module and create the file modules/my_simple_module/config/my_simple_module.js

**Comment by johnpbaldwin** [comment]
I was able to reproduce and grab the screen output to give an example of what I'm seeing. This is on Mac OS X.

`Idle...
Recognized file creation: /Users/john/Documents/Theme/modules/mylinks/html/dialog.html
Uploading file: modules/mylinks/html/dialog.html
File upload complete: modules/mylinks/html/dialog.html
 
Idle...
Recognized file change: /Users/john/Documents/Theme/modules/mylinks/html/dialog.html
Manual intervention required for /Users/john/Documents/Theme/modules/mylinks/html/dialog.html
The file was deleted remotely and updated locally. Unable to determine whether to upload the updated file or delete it.
In order to resolve the conflict you have two choices:
1) Resolve through delete - Deletes the file locally as well.
   This can be achieved by renaming 'modules/mylinks/html/dialog.html' to 'modules/mylinks/html/dialog.html.delete'.
2) Resolve with remote - Download the remote file.
   This can be achieved by renaming 'modules/mylinks/html/dialog.html' to 'modules/mylinks/html/dialog.html.resolve'.
This will cause the system to resolve the conflict.
File download complete: modules/mylinks/html/dialog.html.conflict
 
Idle...
`

**Comment by rywilson-ibm** [comment]
John is the file you're creating empty?
There is a known issue that prevents empty files from being uploaded.  This would explain the behavior that you're seeing.

**Comment by johnpbaldwin** [comment]
Yes. This happens when I create a new file in project mode. The text editor saves an empty file first, then I add content and save.

Also, it appears git may behave in a similar fashion. Creating the empty file, then populating it with data. I'm using git and coordinating between multiple environments, so adding a file in one environment can result in this behavior happening in other environments.

Is there an option to ignore empty files?
